internal/k8s/client.go

```diff
@@ -103,38 +103,27 @@ var _ Client = K8sClient{}
 
 type PortForwarder func(ctx context.Context, restConfig *rest.Config, core apiv1.CoreV1Interface, namespace string, podID PodID, localPort int, remotePort int) (closer func(), err error)
 
-func ProvideK8sClient(ctx context.Context, envOrErr EnvOrError) (Client, error) {
-	env := envOrErr.Env
+func ProvideK8sClient(
+	ctx context.Context,
+	env Env,
+	pf PortForwarder,
+	configNamespace Namespace,
+	runner kubectlRunner,
+	clientLoader clientcmd.ClientConfig) Client {
 	if env == EnvNone {
 		// No k8s, so no need to get any further configs
-		return &explodingClient{err: envOrErr.Err}, nil
+		return &explodingClient{err: fmt.Errorf("Kubernetes context not set")}
```

Comment:
imo it's useful to pass the original error along to this point so that user actually gets some useful info if their k8s stuff fails...?

Reply:
see above comment